force-app/main/default/lwc/datatableMultilineEdit/__tests__/datatableMultilineEdit.test.js

```diff
@@ -0,0 +1,259 @@
+                    return Object.assign({ Id: 'TEST_ID ' + index }, record);
+                });
+                return response;
+            })
+        };
+    },
+    { virtual: true }
+);
+
+jest.mock(
+    '@salesforce/apex',
+    () => {
+        return {
+            refreshApex: jest.fn(() => Promise.resolve())
+        };
+    },
+    { virtual: true }
+);
+
+const DRAFT_VALUES = [
+    {
+        Id: '0031700000pJRRSAA4',
+        FirstName: 'Amy',
+        LastName: 'Taylor',
+        Title: 'VP of Engineering',
+        Phone: '4152568563',
+        Email: 'amy@new_demo.net'
+    },
+    {
+        Id: '0031700000pJRRTAA4',
+        FirstName: 'Michael',
+        LastName: 'Jones',
+        Title: 'VP of Sales',
+        Phone: '4158526633',
+        Email: 'michael@new_demo.net'
+    }
+];
+
+// Sample error for Apex call
+const UPDATE_CONTACTS_ERROR = {
+    body: { message: 'An internal server error has occurred' },
+    ok: false,
+    status: 400,
+    statusText: 'Bad Request'
+};
+describe('c-datatable-multiline-edit', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Prevent data saved on mocks from leaking between tests
+        jest.clearAllMocks();
+    });
+
+    // Helper function to wait until the microtask queue is empty.
+    async function flushPromises() {
+        return Promise.resolve();
+    }
+
+    it('renders six rows in the lightning datatable', async () => {
+        const element = createElement('c-datatable-multiline-edit', {
+            is: DatatableMultilineEdit
+        });
+        document.body.appendChild(element);
+
+        // Emit data from @wire
+        getContacts.emit(mockGetContactList);
+
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+
+        const tableEl = element.shadowRoot.querySelector('lightning-datatable');
+        expect(tableEl.data.length).toBe(mockGetContactList.length);
+        expect(tableEl.data[0].FirstName).toBe(mockGetContactList[0].FirstName);
+    });
+
+    it('updates a record on Save', async () => {
+        const INPUT_PARAMETERS = [{ data: DRAFT_VALUES }];
+
+        // Create initial element
+        const element = createElement('c-datatable-multiline-edit', {
+            is: DatatableMultilineEdit
+        });
+        document.body.appendChild(element);
+
+        // Mock handler for save event
+        const handler = jest.fn();
+        // Add event listener to catch save event
+        element.addEventListener('save', handler);
+
+        // Emit data from @wire
+        getContacts.emit(mockGetContactList);
+
+        // Wait for any asynchronous DOM updates
+        await flushPromises();
+
+        const tableEl = element.shadowRoot.querySelector('lightning-datatable');
```

Comment:
Add a comment that explains that you're simulating a save event on the table

Reply:
Done!